### ISSUE TITLE
Missing semantics for which view option is chosen

### ISSUE THREAD

**Issue by jeryj** [WordPress/gutenberg#63910]
### Description

The View options button in the editor header (Desktop, Tablet, Mobile) shows an icon based on which option is selected, and a checkmark within the dropdown menu to show which is selected. There are no semantics to indicate which option is selected though. It is only visual.

### Step-by-step reproduction instructions

- Using a screenreader
- Navigate to the View button
- The announcement will be something like: "View, Button"
- Select it
- Focus will be on the Desktop option
- Navigate between them and notice it does not change the announcement based on the selected option

### Screenshots, screen recording, code snippet

<img width="443" alt="Screenshot 2024-07-24 at 11 27 02 AM" src="https://github.com/user-attachments/assets/44583557-8307-4473-b864-f2a10ee7dfdf">

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes
